src/read/traits.rs

```diff
@@ -119,6 +119,21 @@ pub trait Object<'data: 'file, 'file>: read::private::Sealed {
     /// Returns an error if the index is invalid.
     fn symbol_by_index(&'file self, index: SymbolIndex) -> Result<Self::Symbol>;
 
+    /// Get the address of symbol named `symbol_name`, if such a symbol exists.
+    fn symbol_address_by_name(&'file self, symbol_name: &str) -> Result<u64> {
```

Comment:
I overlooked the addition of this when reviewing. It didn't really belong in this PR, and I'm not confident that this is an API we want to support because it seems too specialized to me, so I'm going to remove it again.

Reply:
Yes, it seems quite specific, however the function is going to be called in https://github.com/rust-lang/rust/pull/107583/files#diff-6bbf6edd930f26d3b180eb5c250b0e48d8f3c5eb474e3274909ef6ae6f0d1e61R109

Reply:
Since https://github.com/rust-lang/rust/pull/107583 is not landed yet, we can still look for a solution without these new APIs. And as I asked in https://github.com/gimli-rs/object/pull/482#issuecomment-1467810363, we can still use existing APIs and our knowledge of XCOFF to get the work done.